gpcontrib/arenadata_toolkit/arenadata_toolkit--1.0.sql

```diff
@@ -165,3 +165,196 @@ BEGIN
 END$$
 LANGUAGE plpgsql VOLATILE
 EXECUTE ON MASTER;
+
+CREATE FUNCTION arenadata_toolkit.adb_get_relfilenodes(tablespace_oid OID)
+RETURNS TABLE (
+	segindex int2,
+	dbid int2,
+	datoid oid,
+	tablespace_oid oid,
+	relfilepath text,
+	relfilenode oid,
+	reloid oid,
+	size bigint,
+	modified_dttm timestamp without time zone,
+	changed_dttm timestamp without time zone
+)
+EXECUTE ON ALL SEGMENTS
+AS '$libdir/arenadata_toolkit', 'adb_get_relfilenodes'
+LANGUAGE C STABLE STRICT;
+
```

Comment:
Maybe the grants for EXECUTE should be added explicitly for public (instead of implicit behaviour of default [privileges](https://www.postgresql.org/docs/current/ddl-priv.html#PRIVILEGES-SUMMARY-TABLE))?
```sql
GRANT EXECUTE ON FUNCTION arenadata_toolkit.adb_get_relfilenodes(int2, int2, oid, oid, text, oid, oid, bigint, timestamp without time zone, timestamp without time zone) TO public;
```